### ISSUE TITLE
code and parametrisation

### ISSUE THREAD

**Issue by sherpadawan** [wernerjoss/grav-plugin-fullcalendar#16]
1. Javascript should be seperate in another file than template file
2. And a better indentation would be nice to help you.
3. verbose mode (debug) must be a page.header parameter not a inline code parameter ...
4. proxy should be and page.header and/or configuration param as everything else
5. icsfiles shoud be usable as page.header param, user/data/calendars is not very handy but should exists.
I guess parametrisation of fullcalendar instance with plugin load (then no hard coded url in would be in source code) js and php.
Shortcode seems to be powerfull, but page header are much clearer ! What do you think?
Thanks a lot  for your work.

**Comment by wernerjoss** [comment]
@sherpadawan  - thanks for your proposals !
here are my comments, using the same numbering:

1. Javascript is normally located in assets folder, in extra files, thats right, but in this case, twig variables are used inside, so, everything is in the twig template.
2. indentation is done using tabs, looks ok here (using VS Code)
3. verbose mode is not meant to be used/changed by normal users, but only for development, so no need to clutter config with that.
4. CORS proxy is set automatically by intention, upon user request, which is ok, IMHO.
however, a config option to override this with a config parameter by experienced users, might be an improvement
5. no.
I'm a long time Wordpress user and thus used to use shortcodes 😄 .
in addition, shortcodes provide more flexibility, e.g. this way, the plugin can easily be used on different pages with different calendars.

as for README.md cleanup, yes, that makes sense, e.g. by moving some of the comments to the Changelog.

so, as a conclusion, I will check again and incorporate some of your proposals, but not do an 1:1 merge of your pull request.
cheers
Werner

**Comment by sherpadawan** [comment]
Hi !

Thanks for your reply.

Wordpress and Drupal are leading in development culture, but I foudn 
Grav cms easier and faster to manage with page headers.

As you wish, I will continue improving the plugin since my experience 
tought me to make a code more maintanable with a higher parametrisation.

The proposition I did is quite flexible and configuration can be achieve 
both ways (headers and plugin configuration)

Guillaume

Le 20/07/2020 à 08:33, Werner Joss a écrit :

**Comment by sherpadawan** [comment]
There is also a missing feature is rrules management. It is about event 
exceptions that is EXDATE ou EX.. data, but some calendars like ones 
provided by Nextcloud does not export exceptions (in recurrent events) 
as EXDATE or something. So it displays the events chain and another 
chain with the exception ... well there is a bug there.

I tried to spend a few times on this but I did not succeed in managing 
Nextcoud 19 calendar exceptions.

Talk to you later.

Guillaume

Le 20/07/2020 à 08:33, Werner Joss a écrit :
